docs/index.asciidoc

```diff
@@ -0,0 +1,119 @@
+:type: filter
+
+///////////////////////////////////////////
+START - GENERATED VARIABLES, DO NOT EDIT!
+///////////////////////////////////////////
+:version: %VERSION%
+:release_date: %RELEASE_DATE%
+:changelog_url: %CHANGELOG_URL%
+:include_path: ../../../../logstash/docs/include
+///////////////////////////////////////////
+END - GENERATED VARIABLES, DO NOT EDIT!
+///////////////////////////////////////////
+
+[id="plugins-{type}s-{plugin}"]
+
+=== Bytes filter plugin
+
+include::{include_path}/plugin_header.asciidoc[]
+
+==== Description
+
+Parse string representations of computer storage sizes, such as "123 MB" or "5.6gb", into
+their numeric value in bytes.
+
+This plugin understands: 
+
+- bytes ("B")
+- kilobytes ("KB" or "kB")
+- megabytes ("MB", "mb", or "mB")
+- gigabytes ("GB", "gb", or "gB")
+- terabytes ("TB", "tb", or "tB")
+- petabytes ("PB", "pb", or "pB")
+
+[id="plugins-{type}s-{plugin}-examples"]
+==== Examples
+
+
+[cols="<,<",options="header",]
+|=======================================================================
+|Input string |Prefix system | Numeric value in bytes
+|40|`binary` or `decimal`|40
+|40B|`binary` or `decimal`|40
+|40 B|`binary` or `decimal`|40
+|40KB|`binary`|40960
+|40kB|`binary`|40960
+|40KB|`decimal`|40000
+|40.5KB|`binary`|41472
+|40kb|`binary`|5120
+|40Kb|`binary`|5120
+|10 MB|`binary`|10485760
+|10 mB|`binary`|10485760
+|10 mb|`binary`|10485760
+|10 Mb|`binary`|1310720
+|=======================================================================
+
+[source,ruby]
+    filter {
+      bytes {
+        source => "my_bytes_string_field"
+        target => "my_bytes_numeric_field"
+      }
+    }
+
+[id="plugins-{type}s-{plugin}-options"]
+==== Bytes Filter Configuration Options
+
+This plugin supports the following configuration options plus the <<plugins-{type}s-{plugin}-common-options>> described later.
+
+[cols="<,<,<",options="header",]
+|=======================================================================
+|Setting |Input type|Required
+| <<plugins-{type}s-{plugin}-source>> |<<string,string>>|No
+| <<plugins-{type}s-{plugin}-target>> |<<string,string>>|Yes
+| <<plugins-{type}s-{plugin}-prefix_system>> |<<string,string>>|No
+| <<plugins-{type}s-{plugin}-digit_group_separator>> |<<string,string>>|No
+| <<plugins-{type}s-{plugin}-decimal_separator>> |<<string,string>>|No
+|=======================================================================
+
+Also see <<plugins-{type}s-{plugin}-common-options>> for a list of options supported by all
+filter plugins.
+
+&nbsp;
+
+[id="plugins-{type}s-{plugin}-source"]
+===== `source` 
+
+  * Value type is <<string,string>>
+  * Default value is `message`
+
+Name of the source field that contains the storage size
+
+[id="plugins-{type}s-{plugin}-target"]
+===== `target` 
+
+  * Value type is <<string,string>>
+
+Name of the target field that will contain the storage size in bytes
+
+[id="plugins-{type}s-{plugin}-prefix_system"]
+===== `prefix_system` 
```

Comment:
is `prefix_system` a standard term ?  if not can i suggest `conversion_method` or the like ? _(I understand functionality here.. but still don't understand what a prefix_system is)_

Reply:
No, I don't think its standard. I made it up after reading https://en.wikipedia.org/wiki/Metric_prefix and https://en.wikipedia.org/wiki/Binary_prefix. If it doesn't make sense I'm happy to change it; `conversion_method` seems better.